src/core/chmstate_mod.F90

```diff
@@ -25,30 +33,130 @@ module ChmState_Mod
       !---------------------------------------------------------------------
       ! Integers
       !---------------------------------------------------------------------
-      INTEGER           :: nSpecies
-      INTEGER           :: nSpeciesGas
-      INTEGER           :: nSpeciesAero
-      INTEGER           :: nSpeciesDust
-      INTEGER           :: NSpeicesSeaSalt
-      INTEGER, POINTER  :: AeroIndex(:)
-      INTEGER, POINTER  :: GasIndex(:)
-      INTEGER, POINTER  :: DustIndex(:)
-      INTEGER, POINTER  :: SeaSaltIndex(:)
+      INTEGER           :: nSpecies          ! Total Number of Species
+      INTEGER           :: nSpeciesGas       ! Number of Gas Species
+      INTEGER           :: nSpeciesAero      ! Number of Aerosol Species
+      INTEGER           :: nSpeciesDust      ! Number of Dust Species
+      INTEGER           :: NSpeicesSeaSalt   ! Number of SeaSalt Species
+      Integer, POINTER  :: SpeciesIndex(:)   ! Total Species Index
```

Comment:
If array of integers is the goal, probably don't need to be pointers.

Reply:
made them allocatable